zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -8,20 +8,34 @@ mod tests;
 
 use std::{collections::BTreeMap, convert::TryInto, io, sync::Arc};
 
+pub use zcash_history::{V1, V2};
+
 use crate::{
     block::{Block, ChainHistoryMmrRootHash},
-    parameters::{ConsensusBranchId, Network, NetworkUpgrade},
+    orchard,
+    parameters::{Network, NetworkUpgrade},
     sapling,
 };
 
+/// A trait to represent a version of `Tree`.
+pub trait Version: zcash_history::Version {
+    /// Convert a Block into the NodeData for this version.
+    fn block_to_history_node(
+        block: Arc<Block>,
+        network: Network,
+        sapling_root: &sapling::tree::Root,
+        orchard_root: &orchard::tree::Root,
```

Comment:
Instead of ignoring argument this below, should these types be `Option`s?

Reply:
Or we could have different methods for pre-NU5 and post-NU5 trees, so it's easier to review the calling code.

(I don't have a lot of context here, so if this is just internal code, it might be ok.)